### DISCUSSION TITLE
NAFEMS Benchmark T3

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#22964] [Q&A General] [answered]
Hi All

I'm sure I'm doing something super dumb, but I cannot get MOOSE Heat Conduction to agree with the NAFEMS T3 benchmark, initially I thought I was doing something wrong with a Kelvin conversion, but I shifted to C and I see the same problem. 

```
[Mesh]
  type = GeneratedMesh
  dim = 3
  nx = 10000
  ny = 1
  nz = 1
  xmin = 0.0
  xmax = 0.1
  ymin = 0.0
  ymax = 0.01
  zmin = 0.0
  zmax = 0.01
  elem_type = HEX8
[]

[Variables]
  [temp]
    initial_condition = 0
    order = FIRST
  []
[]

[Kernels]
  [heat]
    type = ADHeatConduction
    variable = temp
  []
[]

[Functions]
  [bc_func]
    type = ParsedFunction
    value = '100*(sin(pi*t/40))'
  []
[]

[BCs]
  [dceb]
    type = ADFunctionDirichletBC
    boundary = 'right'
    variable = temp
    function = bc_func
  []
  [ab]
    type = ADDirichletBC
    variable = temp
    boundary = 'left' 
    value = 0 # 0 C
  []
[]

[Kernels]
  [isotropic-heat]
    type = ADHeatConduction
    variable = temp
  []
  [heat-dt]
    type = ADHeatConductionTimeDerivative
    variable = temp
  []
[]

[Materials]
  [density]
    type = ADGenericConstantMaterial
    prop_names = 'density'
    prop_values = '7200.0'
  []
  [copper-heat]
    type = ADHeatConductionMaterial
    temp = temp
    thermal_conductivity_temperature_function = '35.0'
    specific_heat_temperature_function = '440.5'
  []
[]

[Executioner]
  automatic_scaling = true
  type = Transient
  solve_type = NEWTON
  start_time = 0
  end_time = 32
  dt = 1
  petsc_options = '-snes_converged_reason'
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
  line_search = none
[]

[Postprocessors]
  [value_1.25]
    type = PointValue
    variable = temp
    point = '0.08 0.0 0.0'
  []
[]

[Outputs]
  exodus = true
[]
```
Even with a ridiculous mesh density I get 47.7 instead of 36.6, can someone spot my obvious mistake?

**Comment by maxnezdyur** [comment, marked as answer]
You have two kernels of `type = ADHeatConduction`. I think you want to get rid of one of them.

**Reply by makeclean** [reply]
You're quite right, removing that fixes the issue - face palm! @GiudGiud any reason why MOOSE doesn't error or warn when you have a kernel in twice with the same type? Is there a use case where one would want to be able to have the same kernel in with the same definition twice?

**Reply by GiudGiud** [reply]
if you split a momentum diffusion term in molecular + turbulent diffusion I could see one use the same kernel to represent both

Same for multiple reaction terms, a species reaction with multiple species or having multiple sources

The solution here is to use actions instead of specifying your kernels in the input file. 
There s a thermo-mechanical action

Guillaume

**Reply by makeclean** [reply]
In those cases would they literally be the sample kernel instanciated twice with the same variable?

**Reply by GiudGiud** [reply]
The same kernel with the same variable but not the same parameters
